pkg/services/types_test.go

```diff
@@ -85,18 +86,125 @@ func Test_ValidateOrderBy(t *testing.T) {
 		},
 	}
 
-	RegisterTestingT(t)
-
+	g := NewWithT(t)
 	for _, testcase := range tests {
 		tt := testcase
 		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
 			la := NewListArguments(tt.params)
 			err := la.Validate(tt.validParams)
 			if tt.wantErr {
-				Expect(err).To(HaveOccurred())
+				g.Expect(err).To(HaveOccurred())
 			} else {
-				Expect(err).NotTo(HaveOccurred())
+				g.Expect(err).NotTo(HaveOccurred())
+			}
+		})
+	}
+}
+func Test_NewListArguments(t *testing.T) {
+	page := "5"
+	size := "-1"
+	search := "search"
+	overriddenListArgs := &ListArguments{
+		Page:   5,
+		Size:   65500,
+		Search: "search",
+	}
+	type args struct {
+		params url.Values
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want *ListArguments
+	}{
+		{
+			name: "should override default param",
+			args: args{
+				params: url.Values{
+					"page":   []string{page},
+					"size":   []string{size},
+					"search": []string{search},
+				},
+			},
+			want: overriddenListArgs,
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			g.Expect(NewListArguments(tt.args.params)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestListArguments_Validate(t *testing.T) {
+	type fields struct {
+		Page     int
+		Size     int
+		Preloads []string
+		Search   string
+		OrderBy  []string
+	}
+	type args struct {
+		acceptedOrderByParams []string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "should return an error if page is < 0",
+			fields: fields{
+				Page: -1,
+			},
+			args: args{
+				acceptedOrderByParams: getValidTestParams(),
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error if page is < 1",
+			fields: fields{
+				Page: 0,
+			},
+			args: args{
+				acceptedOrderByParams: getValidTestParams(),
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error if there are too many keywords in orderBy",
+			fields: fields{
+				Page:    1,
+				Size:    100,
+				OrderBy: []string{"region desc name owner"},
+			},
+			args: args{
+				acceptedOrderByParams: getValidTestParams(),
+			},
+			wantErr: true,
```

Comment:
We don't have test cases where `wantErr` is false. Either remove it or add a test case with `wantErr == false`